src/lockscreen.js

```diff
@@ -174,38 +176,21 @@ var Lockscreen = GObject.registerClass({
             return;
         if (!this._playbin.hasKey)
             return;
-
-        /* if waiting for the loop to start, leave it */
-        if (this._activeSoundID === 'pending')
-            return;
-
-        /* if the loop is playing, leave it */
-        if (this._activeSoundID !== null && this._activeSoundID !== 'cancel')
-            return;
-
-        SoundServer.getDefault().playAsync('hack-toolbox/lockscreen/active')
-        .then(uuid => {
-            if (this._activeSoundID === 'cancel') {
-                this._activeSoundID = uuid;
-                this._stopActiveSound();
-                return;
-            }
+        const playPromise = SoundServer.getDefault().playAtBusNameAsync(
+            'hack-toolbox/lockscreen/active', this.toolbox.target_app_id);
+        playPromise.then(uuid => {
             this._activeSoundID = uuid;
         });
-        this._activeSoundID = 'pending';
     }
 
     _stopActiveSound() {
-        if (this._activeSoundID === null)
-            return;
-        if (this._activeSoundID === 'cancel')
-            return;
-        if (this._activeSoundID === 'pending') {
-            this._activeSoundID = 'cancel';
-            return;
+        if (this._activeSoundID !== null) {
+            SoundServer.getDefault().stopAtBusName(
+                this._activeSoundID,
+                this.toolbox.target_app_id
```

Comment:
Here (and similar places) it would only need to access it own `target-app-id` property.